gpii/node_modules/lifecycleManager/src/LifecycleManagerSession.js

```diff
@@ -71,18 +72,90 @@ var gpii = fluid.registerNamespace("gpii");
         }
     });
 
+    gpii.lifecycleManager.session.setAutoSavePrefs = function (existingAutoSavePrefs, preferences, path) {
+        var prefsValue = fluid.get(preferences, path);
+        if (prefsValue) {
+            fluid.set(existingAutoSavePrefs, path, prefsValue);
+        }
+        return existingAutoSavePrefs;
+    };
+
+    /**
+     * Filter the preferences by metadata defined in the preferences set. Only the preferences that are allowed to be
+     * autosaved will be returned.
+     * @param {Object} preferences - The updated preferences.
+     * @param {Object} metadata - The metadata that defined in the prefs set.
+     * @return {Object} - The preferences to be autosaved.
```

Comment:
Make sure to document that the type is really {Object|Undefined} with a special interpretation